default/python/AI/EspionageAI.py

```diff
@@ -54,7 +65,7 @@ def colony_detectable_by_empire(planet_id, species_name=None, empire_id=ALL_EMPI
         species_name = planet.speciesName
 
     # in case we are checking about aborting a colonization mission
-    if empire_id == fo.empireID() and planet.ownedBy(empire_id):
+    if empire == fo.empireID() and planet.ownedBy(empire):
```

Comment:
This is an issue:
`1 == fo.empireID()` -> True
`[1] == fo.empireID()` -> False

Reply:
As has now been clarified in the docstrings, there isn't any use-case for intentionally passing a list containing only the current empire ID.  The only way that could even come up accidentally is in some sort of test multiplayer game in which the only empire is an AI empire, and in that case we would want the test to fall through as it currently does.

Reply:
> As has now been clarified in the docstrings, there isn't any use-case for intentionally passing a list containing only the current empire ID.

And passing int as current empire id is valid use case?  Docstring is invalid, since it was not updated after argument type was changed.

My complain is that int as argument and list with single argument with same int have different behavior.  It is not clear and it is not described in docstring.   When I see `int | list[int]` type, I assume that it is shortcut to remove unnecessary list wrapping:   `x(n) == x([n]) `.

Reply:
> And passing int as current empire id is valid use case? 

There is a fair bit specifically about that in both the docstring and some in a code comment.  I had added a fair bit of that specifically in response to your earlier comments, as I pretty clearly indicated in my comment above.  Are you sure you have read the current code?

> Docstring is invalid, since it was not updated after argument type was changed.

While some sort of relative-timing concern like that might be good reason to examine something closely, it's not much of an argument for asserting something is invalid.   I rather suspect that you simply misread/misinterpreted something.  

The docstring does in fact look valid to me.  If you wish to assert it is invalid, please be quite specific about what you think is invalid and why it would actually be invalid (not simply why you are suspicious).

> My complain is that int as argument and list with single argument with same int have different behavior. It is not clear and it is not described in docstring. 

To the contrary, it is described in the docstring, quite explicitly (and to my eyes, clearly).  Were you somehow still looking at an old page of code even after I said I explained all this in the docstring?

>When I see int | list[int] type, I assume that it is shortcut to remove unnecessary list wrapping: x(n) == x([n]).

I think I must suggest you reread the docstring, and beware of over-reliance on assumptions.